komodo/lint_maturity.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import sys
+import yaml
+from packaging.version import Version, parse
+
+
+INVALID_TAGS = {
+    "a": ["Invalid"],
+    "b": ["a", "Invalid"],
+    "rc": ["a", "b", "Invalid"],
+    "None": ["a", "b", "rc", "Invalid"],
+}
+
+
+def get_parser():
+    parser = argparse.ArgumentParser(description=("Lint the maturity of packages."))
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument(
+        "--release_file",
+        type=lambda arg: arg
+        if os.path.isfile(arg)
+        else parser.error("{} is not a valid file".format(arg)),
+        help="Release file",
+    )
+    group.add_argument(
+        "--release_folder",
+        type=lambda arg: arg
+        if os.path.isdir(arg)
+        else parser.error("{} is not a valid directory".format(arg)),
+        help="File with all package tags named as release version.",
+    )
+    return parser
+
+
+def define_release_version(release_file):
+    try:
+        Version(release_file)
+        release_version = (
+            "None"
+            if parse(release_file).pre is None
+            else str(parse(release_file).pre[0])
+        )
+    except:
+        release_version = "Invalid"
+    return release_version
+
+
+def run(files_to_lint):
+    system_exit=""
+    for file_to_lint in files_to_lint:
+        release_file_basename = os.path.basename(file_to_lint)
+        release_file_clean = os.path.basename(file_to_lint)[
+            0 : (len(release_file_basename) - 9)
+        ]  # remove -pyXX.yml
+        release_version = define_release_version(release_file_clean)
+
+        if release_version == "Invalid":
+            system_exit=system_exit+release_file_basename+ " is incompatible with version naming \n"
+        elif release_version != "Invalid":
+            dict_tag_maturity = {"a": {}, "b": {}, "rc": {}, "None": {}, "Invalid": {}}
+
+            with open(file_to_lint) as rf:
+                package_version = yaml.safe_load(rf)
+
+            for key, value in package_version.items():
+                release_version_package = define_release_version(value)
+                dict_tag_maturity[release_version_package][key] = value
+
+            count_tag_maturity = {
+                key: len(dict_tag_maturity[key]) for key in dict_tag_maturity
+            }
+            count_tag_invalid = sum(
+                count_tag_maturity[key] for key in INVALID_TAGS[release_version]
+            )
+            
+            system_exit=system_exit+release_file_basename+" has "+str(count_tag_invalid)+" packages with invalid maturity tag. \n"
+            if count_tag_invalid > 0:
+                for key in INVALID_TAGS[release_version]:
+                    if len(dict_tag_maturity[key]) > 0:
+                        system_exit=system_exit+"         Tag "+ key+ " packages: "+ str(dict_tag_maturity[key])+'\n'
+
+    raise SystemExit(system_exit)
```

Comment:
This function does a lot, and I would consider splitting it up into several smaller functions, specifically regarding the file reading. A side effect is that it might make it easier to test as well, and easier to understand what is going on.

Reply:
I tried to review and it was indeed possible to split and make more readable. Thanks for highlighting this.